x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportDeleteInferenceEndpointAction.java

```diff
@@ -0,0 +1,173 @@
+                    )
+                );
+                return;
+            } else if (request.isForceDelete() == false
+                && endpointIsReferencedInPipelines(state, request.getInferenceEndpointId(), listener)) {
+                    return;
+                }
+
+            var service = serviceRegistry.getService(unparsedModel.service());
+            if (service.isPresent()) {
+                service.get().stop(request.getInferenceEndpointId(), listener);
+            } else {
+                listener.onFailure(
+                    new ElasticsearchStatusException(
+                        "No service found for this inference endpoint " + request.getInferenceEndpointId(),
+                        RestStatus.NOT_FOUND
+                    )
+                );
+            }
+        }).<Boolean>andThen((listener, didStop) -> {
+            if (didStop) {
+                modelRegistry.deleteModel(request.getInferenceEndpointId(), listener);
+            } else {
+                listener.onFailure(
+                    new ElasticsearchStatusException(
+                        "Failed to stop inference endpoint " + request.getInferenceEndpointId(),
+                        RestStatus.INTERNAL_SERVER_ERROR
+                    )
+                );
+            }
+        })
+            .addListener(
+                masterListener.delegateFailure((l3, didDeleteModel) -> masterListener.onResponse(AcknowledgedResponse.of(didDeleteModel)))
+            );
+    }
+
+    private static boolean endpointIsReferencedInPipelines(
+        final ClusterState state,
+        final String inferenceEndpointId,
+        ActionListener<Boolean> listener
+    ) {
+        Metadata metadata = state.getMetadata();
+        if (metadata == null) {
+            listener.onFailure(
+                new ElasticsearchStatusException(
+                    " Could not determine if the endpoint is referenced in a pipeline as cluster state metadata was unexpectedly null. "
+                        + "Use `force` to delete it anyway",
+                    RestStatus.INTERNAL_SERVER_ERROR
+                )
+            );
+            // Unsure why the ClusterState metadata would ever be null, but in this case it seems safer to assume the endpoint is referenced
+            return true;
+        }
+        IngestMetadata ingestMetadata = metadata.custom(IngestMetadata.TYPE);
+        if (ingestMetadata == null) {
+            logger.debug("No ingest metadata found in cluster state while attempting to delete inference endpoint");
+        } else {
+            Set<String> modelIdsReferencedByPipelines = InferenceProcessorInfoExtractor.getModelIdsFromInferenceProcessors(ingestMetadata);
+            if (modelIdsReferencedByPipelines.contains(inferenceEndpointId)) {
+                listener.onFailure(
+                    new ElasticsearchStatusException(
+                        "Model "
```

Comment:
Should this say `Inference endpoint` instead of `Model`?

Reply:
I think I'll do a follow up PR that more thoroughly replaces `model` in the module